crates/flox/src/commands/mod.rs

```diff
@@ -435,13 +437,49 @@ impl AdditionalCommands {
             AdditionalCommands::Unsubscribe(args) => args.handle(flox).await?,
             AdditionalCommands::Channels(args) => args.handle(flox)?,
             AdditionalCommands::History(args) => args.handle(flox).await?,
-            AdditionalCommands::PrintDevEnv(args) => args.handle(config, flox).await?,
-            AdditionalCommands::Shell(args) => args.handle(config, flox).await?,
+            AdditionalCommands::PrintDevEnv(args) => {
+                deprecate_command("print-dev-env");
+                args.handle(config, flox).await?
+            },
+            AdditionalCommands::Shell(args) => {
+                deprecate_command("shell");
+                args.handle(config, flox).await?
+            },
         }
         Ok(())
     }
 }
 
+fn check_deprecated_commands(commands: &Commands) {
+    match commands {
+        Commands::Development(_) => { /* none deprecated */ },
+        Commands::Sharing(_) => { /* none deprecated */ },
+        Commands::Additional(additional_commands) => match additional_commands {
+            AdditionalCommands::Build(_) => deprecate_command("build"),
```

Comment:
if build is deprecated shouldn't publish be as well

Reply:
added publish and init-package